### ISSUE TITLE
AD Format

### ISSUE THREAD

**Issue by JCBNim** [PharmGKB/PharmCAT#168]
<img width="921" alt="Capture d’écran 2024-01-12 à 11 30 28" src="https://github.com/PharmGKB/PharmCAT/assets/66789838/402e5d40-22bf-43c8-8d88-fb7898a23c82">

* **Do you want to request a *feature* or report a *bug*?** a bug

* **What is the current behavior?** 
VCF
	Unexpected AD format number: '2'. Treating number as '.' and ignoring AD field.

* **If the current behavior is a bug, please provide the steps to reproduce and, if possible, your example input data** via a [Gist](https://gist.github.com) or similar.

* **What is the expected behavior?**

* **What is the motivation / use case for changing the behavior?**

* **Please tell us about your environment:**
  
  - PharmCAT Version: ??? 
  - JDK Version: [ JDK8 | OpenJDK | etc ]
  - Environment: [ Windows | macOS | Linux distro | etc ] macOS

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, e.g. stackoverflow, gitter, etc.)
* Good morning, the issue about the AD format linked to the VCF I use. The HTML report doesn't run for some genes perhaps due to the AD format used in the VCF given by SophiaGenetics. Could you please help us.  Thanks.  JC BOYER

**Comment by JCBNim** [comment]
Hello,
I want to know if the "AD" format requires an integer or if a number with decimals is possible. 
The Pharmcat software tells me the following message with the numbers detailed in the screenshot: 
"VCF  Unexpected AD format number: '2'. Treating number as '.' and ignoring AD field.

The attached file is a screenshot of part of my VCF generated by "Sophiagenetics DDM" where you can see the AD format

**Comment by markwoon** [comment]
> I want to know if the "AD" format requires an integer or if a number with decimals is possible.

I'm going to answer your question with another question:  do you understand what the AD field represents?  If not, refer to the "AD and DP" section here: https://gatk.broadinstitute.org/hc/en-us/articles/360035531692-VCF-Variant-Call-Format

TL;DR - Since it represents the # of reads, it must be an integer. 


> The Pharmcat software tells me the following message with the numbers detailed in the screenshot:
> "VCF Unexpected AD format number: '2'. Treating number as '.' and ignoring AD field.

This error is not talking about the VALUE you have, it refers to the INFO header for the AD field.

I will improve the error message to: `INFO header for AD has unexpected number (2). Expecting 'R'. Treating number as '.' and ignoring AD field.`  

See section 1.2.2 in the VCF spec for details.

TL;DR - You should have an AD value for each possible allele, which is represented by "R".
